app/services/requirements/video_embed_checker.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+module Requirements
+  class VideoEmbedChecker
+    YOUTUBE_PREFIX = "https://www.youtube.com/watch"
+
+    def pre_embed_issues(title, url)
+      issues = []
+
+      if title.blank?
+        issues << Issue.new(:video_embed_title, :blank)
+      end
+
+      if url.blank?
+        issues << Issue.new(:video_embed_url, :blank)
+      elsif !url.start_with?(YOUTUBE_PREFIX)
```

Comment:
I see this matches the hint text which is actually a rather simplified approach to what is accepted as video input.

We accept both youtube.com and youtu.be urls. And youtube.com urls require a query string parameter of v. 

It doesn't seem too problematic if the JS is more forgiving than this check, so I think it's probably fine to drop youtu.be URLs (pending us getting feedback) but I think it should match in strictness to check the query string so that it will accept the same.

Reply:
You caught me before I had the chance to ask :-). This aspect was missing from the card, and the way the hint text is framed makes it ambiguous as a feature.

My suggestion is actually a bit different to yours: we should support youtu.be URLs as they're produced by YouTube, so it's reasonable a user might try to put one in, and the hint text should be changed to something more generic like 'be a valid YouTube URL'.

When it comes to validation, we could elongate the prefix with `?v=`, but I don't see the value in making the check more complex than simply checking one or more prefixes, as it's unlikely users will be typing the URL manually. How does that sound?

Reply:
I think the hint text is quite good in how explicit and succinct it is, so that could be tricky to tweak for generic.

I don't mind if we support youtu.be as well though.

Yeah I think it's ok to include the query string as part of the prefix check. I did have hesitations as, of course, the v could be anywhere in the query string - but YouTube seems to make a concerted effort to have that as the first param so seems unlikely to not be that.

Reply:
OK. Had a chat with @benhazell and he agrees the content is fine as-is. I've come round to making the check more complex :-).